dalm/datasets/reading_comprehension_generation/regex_based.py

```diff
@@ -0,0 +1,1243 @@
+                text=context_wo_title, domain=domain_name, title=title, sents=copy.deepcopy(sents)
+            )
+            # mined_num is the number of mined examples per task type
+
+        # create the reading comprehension text
+        read_compre, count_dict = overall_cls.format_recomprehension(copy.deepcopy(overall_entry))
+        # count_dict includes the number of comprehension tasks per task type
+        # you may use `mined_num` and `count_dict` for data analysis
+        read_compre_list.append(read_compre)
+
+    return {"read_compre": read_compre_list, "file_name": entry["file_name"]}
+
+
+class RC:
+    def __init__(self, general_spm, domain_spm):
+        self.inited_type_map = {}
+
+        for type in TYPES:
+            type_cls = type_map.cls_dic[type]()
+            type_cls.init_spm(general_spm, domain_spm)
+            self.inited_type_map[type] = type_cls
+
+        self.overall_cls = type_map.cls_dic["overall"]()
+        self.overall_cls.init_spm(general_spm, domain_spm)
+
+        # to chunk text to sentences
+        self.segmenter = Segmenter(language="en", clean=False)
+
+    def generate(self, entry):
+        # NOTE: if the context has no title, use the following code:
+        title = None
+        context_wo_title = entry["text"]
+
+        # truncate the context to meet the max_seq_len
+        # context_wo_title = overall_cls.truncate_sentence(context_wo_title, max_len=overall_cls.max_seq_len-200)
+        context_wo_title_list = self.overall_cls.truncate_sentence(
+            context_wo_title, max_len=self.overall_cls.max_seq_len - 200
+        )
+
+        read_compre_list = []
+        for context_wo_title in context_wo_title_list:
+            sents = self.segmenter.segment(context_wo_title)
+            overall_entry = {"text_id": entry["text_id"]}
+            for type in TYPES:
+                type_cls = self.inited_type_map[type]
+                overall_entry[type], mined_num = type_cls.mine(
+                    text=context_wo_title, title=title, sents=copy.deepcopy(sents)
+                )
+
+        # create the reading comprehension text
+        read_compre, count_dict = self.overall_cls.format_recomprehension(copy.deepcopy(overall_entry))
+        # count_dict includes the number of comprehension tasks per task type
+        # you may use `mined_num` and `count_dict` for data analysis
+        read_compre_list.append(read_compre)
+
+        return {"read_compre": read_compre_list, "file_name": entry["file_name"]}
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--input_dir", type=str, help="directory of the input raw texts", default="./data")
+    parser.add_argument(
+        "--output_dir", type=str, help="directory of the output reading comprehension texts", default="./output"
+    )
+    parser.add_argument(
+        "--ori_spm_path", type=str, help="path of the original sentencepiece model", default="./tokenizers/general.spm"
```

Comment:
Could the code auto-download that somehow?  If not, it seems like something that is good to add to the README example.  Ditto for `--domain_spm_path` and `--domain_tokenizer_training_text`

Reply:
Good one. But this domain tokenizer needs to get trained with user data.